woodwork/tests/accessor/test_statistics.py

```diff
@@ -80,7 +80,8 @@ def test_accessor_replace_nans_for_mutual_info():
         'str_no_nan': pd.Series(['test', 'test2', 'test2', 'test']),
         'dates': pd.Series(['2020-01-01', None, '2020-01-02', '2020-01-03'])
     })
-    df_nans.ww.init()
+    df_nans.ww.init(logical_types={"bools": "Categorical", "str":
+                                   "Categorical", "str_no_nan": "Categorical"})
```

Comment:
Is this because we would now infer these column types differently than we did before?

Reply:
Yeah, it's because `_replace_nans_for_mutual_info` has some if conditionals that include things like `column.is_numeric` and `column.is_categorical`.  Those end up taking different branches after this change because the columns mentioned in the line you highlight are no longer inferred as categorical.  Updating the sample data to be inferred differently felt like a pain in the butt because I'd also have to update all the lines further down in the test case.  On the other hand, all we really care about testing is that the internal logic of the `_replace_nans_for_mutual_info` works correctly.  It seemed just as effective to do that by forcing certain logical types with `ww.init(logical_types=...)` than by having those types inferred.